lib/indocker/launchers/configuration_deployer.rb

```diff
@@ -423,23 +423,45 @@ def sync_artifacts(clonner, artifact_servers)
               logger: @logger
             )
 
-            @logger.info("Pulling git artifact  #{artifact.name.to_s.green} for #{server.user}@#{server.host}")
-            result = clonner.clone(session, artifact.repository)
+            if artifact.is_a?(Indocker::Artifacts::Git)
```

Comment:
we should not have `IF` condition for artifact in configuration deployer. ConfigurationDeployer is very high level class and should not know about different types of artifacts. Please move this to a separate class and try to refactor so we don't have `IF` condition.